src/Models/ApplicantModel.php

```diff
@@ -561,17 +568,58 @@ public function appendToAssessmentNotes(int $id, string $assessmentNote): bool
 
     public function updateEdaid(int $id, int $edaid): bool
     {
-        $query = $this->db->prepare('UPDATE `applicants_additional` SET edaid=:edaid WHERE id=:id');
+        $query = $this->db->prepare('UPDATE `applicants_additional` SET edaid=:edaid WHERE `id` = :id;');
         return $query->execute(["id" => $id, "edaid" => $edaid]);
     }
 
     public function updateGithubUsername(int $id, string $githubUsername): bool
     {
-        $query = $this->db->prepare('UPDATE `applicants_additional` SET githubUsername=:githubUsername
-     WHERE id=:id');
+        $query = $this->db->prepare('UPDATE `applicants_additional` SET githubUsername =: githubUsername
+     WHERE `id` = :id');
         return $query->execute(["id" => $id, "githubUsername" => $githubUsername]);
     }
+    public function toggleLockField(int $id, string $fieldName)
+    {
+        switch ($fieldName) {
+            case 'githubUsername':
+                $query = $this->db->prepare(
+                    'SELECT `githubUsernameLocked` 
+                    FROM `applicants_additional` 
+                    WHERE `id` = :id'
+                );
+                $query->execute(["id" => $id]);
+                $result = $query->fetch();
+                $locked = $result['githubUsernameLocked'] ? 0 : 1;
+                $sql = 'UPDATE `applicants_additional` SET `githubUsernameLocked` = :locked WHERE `id` = :id';
+                break;
```

Comment:
the code in each of these case statements is identical except for the field name, could be made into a function with the fieldname as a param, this would dramatically reduce duplicate code